File: Fundamentals/Dictionaries - Exercise/courses.py
commands = input()
courses = dict()
while commands != "end":
    commands = commands.split(" : ")
    course_name = commands[0]
    student_name = commands[1]
    if course_name not in courses.keys():
        courses[course_name] = list()
    courses[course_name].append(student_name)

    commands = input()
for course, list_of_student in courses.items():
    count_of_students = len(list_of_student)
    print(f"{course}: {count_of_students}")
    for student in list_of_student:
        print(f"-- {student}")

File: JavaFundamentals/src/MapsLambdaAndStreamAPI/Exercise/Courses.java
package MapsLambdaAndStreamAPI.Exercise;

import java.util.*;

public class Courses {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        Map<String, List<String>> courses = new LinkedHashMap<>();
        String input;
        while (!(input = scanner.nextLine()).equals("end")) {
            String[] data = input.split(" : ");

            String courseName = data[0];
            String studentName = data[1];

            courses.computeIfAbsent(courseName, k -> new ArrayList<>()).add(studentName);
        }

        for (Map.Entry<String, List<String>> course : courses.entrySet()) {
            System.out.printf("%s: %d%n", course.getKey(), course.getValue().size());
            for (String student : course.getValue()) {
                System.out.printf("-- %s%n", student);
            }
        }
    }
}
